Sources/SotoCore/Credential/ConfigFileLoader.swift

```diff
@@ -0,0 +1,318 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Soto for AWS open source project
+//
+// Copyright (c) 2017-2020 the Soto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Soto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import struct Foundation.UUID
+import INIParser
+import Logging
+import NIO
+#if os(Linux)
+import Glibc
+#else
+import Foundation.NSString
+#endif
+
+public enum ConfigFile {
+    public static let defaultCredentialsPath = "~/.aws/credentials"
+    public static let defaultProfileConfigPath = "~/.aws/config"
+    public static let defaultProfile = "default"
+}
+
+/// Load settings from AWS credentials and profile configuration files
+/// https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
+enum ConfigFileLoader {
+    /// Specific type of credentials loaded from disk
+    enum SharedCredentials: Equatable {
+        case staticCredential(credential: StaticCredential)
+        case assumeRole(roleArn: String, sessionName: String, region: Region?, sourceCredential: StaticCredential)
+        case credentialSource(roleArn: String, source: CredentialSource)
```

Comment:
`credentialSource` will require `sessionName` and `region`

Reply:
Right now it throws an exception, as it is not yet supported. Not sure which parameters will be needed, I'd like to leave that for another PR.

Reply:
Ok we'll get this one in. I can add the additional stuff. I'll create a PR you can review.